api/src/main/java/com/velocitypowered/api/proxy/server/PingOptions.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2018-2023 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.proxy.server;
+
+import com.velocitypowered.api.network.ProtocolVersion;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import net.kyori.adventure.builder.AbstractBuilder;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents the characteristics with which a ping to a Minecraft server is simulated.
+ * This class is immutable.
+ *
+ * @since 3.2.0
+ */
+public final class PingOptions {
+  private final ProtocolVersion protocolVersion;
+  private final long timeout;
+
+  private PingOptions(final Builder builder) {
+    this.protocolVersion = builder.protocolVersion;
+    this.timeout = builder.timeout.toMillis();
+  }
+
+  /**
+   * The version of Minecraft from which ping will be emulated.
+   *
+   * @return the emulated Minecraft version
+   */
+  public ProtocolVersion getProtocolVersion() {
+    return this.protocolVersion;
+  }
+
+  /**
+   * The maximum timeout duration for receiving the {@link ServerPing}.
+   *
+   * @return the server ping timeout in milliseconds
+   */
+  public long getTimeout() {
+    return this.timeout;
+  }
+
```

Comment:
Some users prefer the convention `PingOptions.builder().[...].build().ping(server)` instead of `server.ping(PingOptions.builder().[...].build())`. Can we add that helper method that forwards the call to the existing ping member in `RegisteredServer`? I must say this is a big nit, however.

Reply:
I'm not so sure about adding that method, I don't see much need to add it, besides that in Velocity there is no class that is built by builder that behaves that way

Reply:
I agree :) Let's ignore it then.

Reply:
I agree, let's keep it this way :)